### PR TITLE
VZ-5744 Alternative testing approach for prometheus E2E tests

### PR DESCRIPTION
**Pull request by anajosep**
# Description

Contains E2E tests for prometheus pre and post upgrade. Tests validate if an app deployed with an ingress trait, sends metric pre and post upgrade. Also tests the system metrics are exported to prometheus pre and post upgrade.

Fixes VZ-5744

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: ci/upgrade-paths/Jenkinsfile [hunk 1]
```diff
@@ -1317,6 +1329,69 @@ def runGinkgoVerify(testSuitePath, clusterDumpDirectory, skipDeploy, skipUndeplo
...
+                export DUMP_KUBECONFIG="${KUBECONFIG_DIR}/${count}/kube_config"
+                export DUMP_DIRECTORY="${clusterDumpDirectory}${clusterName}"
+                cd ${GO_REPO_PATH}/verrazzano/tests/e2e
+                ginkgo -v --keep-going --no-color ${GINKGO_REPORT_ARGS} -tags="${params.TAGGED_TESTS}" --focus-file="${params.INCLUDED_TESTS}" --skip-file="${params.EXCLUDED_TESTS}" ${testSuitePath}/... -- --skipDeploy=${skipDeploy} --skipUndeploy=${skipUndeploy} --skipVerify=${skipVerify} --namespace=${namespace}
+            """
+        }
+    }
+}
+
+def runGinkgoVerifySerial(testSuitePath, clusterDumpDirectory, skipDeploy, skipUndeploy, skipVerify, namespace) {
```
Comment (pshirodkar-orcl) [inline ci/upgrade-paths/Jenkinsfile:1368]: Please remove this unreferenced method.

File: tests/e2e/metrics/deploymetrics/deploymetrics_test.go [hunk 2]
```diff
@@ -52,7 +67,7 @@ func deployMetricsApplication() {
 		nsLabels := map[string]string{
 			"verrazzano-managed": "true",
 			"istio-injection":    istioInjection}
-		return pkg.CreateNamespace(testNamespace, nsLabels)
+		return pkg.CreateNamespace(namespace, nsLabels)
```
[the remark this answers is not in the document]
Comment (anajosep) [reply tests/e2e/metrics/deploymetrics/deploymetrics_test.go:70]: These are system component metrics and  is covered in `metrics/syscomponents` test suite.
In `ci/upgrade-paths/Jenkinsfile` These validations are included under 
```
stage ('Verify Metrics') {
                    steps {
                        runGinkgoRandomizeParallel('metrics/syscomponents')
                    }
                }
```
So the verification of system metrics is covered and is not required to be covered again.

File: ci/upgrade-paths/Jenkinsfile (hunk 1 above)
Comment (anajosep) [reply ci/upgrade-paths/Jenkinsfile:1368]: Ok. I will remove this unused method.

File: tests/e2e/metrics/deploymetrics/deploymetrics_suite_test.go [hunk 3]
```diff
@@ -11,9 +11,17 @@ import (
...
+var skipDeploy bool
+var skipUndeploy bool
+var namespace string
+var skipVerify bool
 var istioInjection string
 
 func init() {
+	flag.BoolVar(&skipDeploy, "skipDeploy", false, "skipDeploy skips the call to install the application")
+	flag.BoolVar(&skipUndeploy, "skipUndeploy", false, "skipUndeploy skips the call to install the application")
+	flag.StringVar(&namespace, "namespace", "deploymetrics", "namespace is the app namespace")
```
[the remark this answers is not in the document]
Comment (anajosep) [reply tests/e2e/metrics/deploymetrics/deploymetrics_suite_test.go:23]: I didn't want to change the default behaviour of the existing test. Currently it uses `deploymetrics` as the namespace value. Changing it to use `generatedNamespace` might have some impacts. Let me try it out and if it works as expected, I can make that change.

**Review by pshirodkar-orcl — APPROVED** [review]

**Pull request merged**